feature-libs/cart/base/components/added-to-cart-toast/added-to-cart-toast.component.ts

```diff
@@ -0,0 +1,192 @@
+
+  toastContainerBaseClass = 'toast-container';
+
+  scrollEventUnlistener: () => void;
+
+  @HostBinding('className') baseClass: string;
+
+  cartToasts$: Observable<CartToastItem[]> =
+    this.addedToCartToastService.getToasts();
+
+  constructor(
+    protected winRef: WindowRef,
+    protected renderer: Renderer2,
+    protected cd: ChangeDetectorRef,
+    protected addedToCartToastService: AddedToCartToastComponentService,
+    protected activeCartService: ActiveCartFacade,
+    protected el: ElementRef
+  ) {}
+
+  init(): void {
+    if (!this.customClass) {
+      this.customClass = 'cx-added-to-cart-toast';
+    }
+
+    this.baseClass = `${this.customClass}`;
+    this.toastContainerClass = this.toastContainerBaseClass;
+    this.headerElement = this.winRef.document.querySelector('header');
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  addToast(event: CartUiEventAddToCart) {
+    this.setOffsetHeight();
+
+    this.activeCartService
+      .getLastEntry(event.productCode)
+      .pipe(
+        filter(Boolean) as OperatorFunction<OrderEntry | undefined, OrderEntry>,
+        take(1)
+      )
+      .subscribe((entries) => {
+        let toastContainerClass = `${this.toastContainerBaseClass} `;
+        const toastItem = this.addedToCartToastService.addToast(
+          event.quantity,
+          entries,
+          toastContainerClass
+        );
+
+        this.cd.detectChanges();
+
+        // trigger enter animations with class change
+        toastItem.baseClass = this.getToastStyles();
+
+        if (this.scrollEventUnlistener) {
+          this.scrollEventUnlistener();
+        }
+
+        // wait for enter animation end
+        setTimeout(() => {
+          this.triggerScrollEvent();
+        }, 500);
+
+        // trigger slide out animation after timeout
+        toastItem.timeoutRef = setTimeout(() => {
+          this._closeToast(toastItem);
+        }, this.timeout);
+      });
+  }
+
+  setOffsetHeight() {
+    if (this.headerElement) {
+      this.el.nativeElement.style.setProperty(
+        '--cx-added-to-cart-toast-header-offset',
+        `${this.headerElement?.offsetHeight}px`
+      );
+    }
+  }
+
+  triggerScrollEvent(): void {
+    this.scrollEventUnlistener = this.renderer.listen(
+      this.winRef.nativeWindow,
+      'scroll',
+      this._setPosition
+    );
+  }
+
+  isHeaderInView(): boolean {
+    if (!this.headerElement) {
+      return false;
+    }
+    const headerBounding = this.headerElement.getBoundingClientRect();
+    return headerBounding.bottom >= 0;
+  }
+
+  isHeaderFixed() {
+    if (!this.headerElement) return false;
+    const headerPosition = this.headerElement.style.position;
+    return headerPosition === 'fixed';
+  }
+
+  closeToast(toastItem: CartToastItem) {
+    if (toastItem.timeoutRef) {
+      clearTimeout(toastItem.timeoutRef);
+    }
+    this._closeToast(toastItem);
+  }
+
+  _closeToast = (toastItem: CartToastItem) => {
```

Comment:
why the underscore?

Reply:
[2022/06/07]  - Answer: because of a name clash.  Is there a way to handle this in a less confusing way?